Modules/Core/Common/include/itkConicShellInteriorExteriorSpatialFunction.hxx

```diff
@@ -107,10 +107,10 @@ ConicShellInteriorExteriorSpatialFunction<VDimension, TInput>::PrintSelf(std::os
   }
   os << "]" << std::endl;
 
-  os << indent << "DistanceMin: " << m_DistanceMin << std::endl;
-  os << indent << "DistanceMax: " << m_DistanceMax << std::endl;
-  os << indent << "Epsilon: " << m_Epsilon << std::endl;
-  os << indent << "Polarity: " << m_Polarity << std::endl;
+  os << indent << "DistanceMin: " << m_DistanceMin << std::endl
+     << indent << "DistanceMax: " << m_DistanceMax << std::endl
```

Comment:
This change looks pointless.

Reply:
It's basically what the PR is doing: avoiding to mention the output stream variable repetitively. 🤷  

The idea of this PR is that rather than Pascal style `WriteLine(str)` statements (which print one line at a time), we could insert multiple lines into one and the same stream, by a single chain of `<<` insertions.

Reply:
But for the implementation of all those `PrintSelf` member functions, it might be even nicer to have some `PrintData` function that would write an arbitrary number of name-value pairs to the specified `os`, inserting the specified `indent` before each name-value-pair, e.g.: 

    PrintData(os, indent, { "Epsilon", m_Epsilon }, { "Polarity", m_Polarity }, { "DistanceMin", m_DistanceMin });

Could be implemented as a variadic template, possibly as a member function of `itk::LightObject`. What do you think?

Reply:
This pattern appears a lot:
```cpp
os << indent << "DistanceMin: " << m_DistanceMin << std::endl;
```
we could have a macro which makes this less prone to errors:
```cpp
#define PrintMember(name) os << indent << #name ": " << m_##name << std::endl ITK_MACROEND_NOOP_STATEMENT
```
which would be invoked like this: `PrintMember(DistanceMin);`.

Reply:
Update: with my latest amend, I excluded the `os <<` insertions, which are typically from inside PrintSelf member functions. For PrintSelf a different approach may be more useful, as you suggested. Using a macro could be helpful, although I'd rather not hide away `os` and `indent` behind a macro definition. 

I'm thinking of introducing a `Printer` class, which would support chaining multiple `PrintMember` calls:

    Printer(os, indent).PrintMember("Epsilon", m_Epsilon).PrintMember("Polarity", m_Polarity);

And a `PrintMemberMacro(name)` macro, to shorten the above:

    Printer{os, indent}.PrintMemberMacro(Epsilon).PrintMemberMacro(Polarity);

Note that in some cases, the macro won't work anyway, for example `os << indent << "Pointer: " << static_cast<void *>(m_ImportPointer) << std::endl` at https://github.com/InsightSoftwareConsortium/ITK/blob/da36517a51d92cc38a98c544348115af124e5f8a/Modules/Core/Common/include/itkImportImageContainer.hxx#L216

Reply:
This might not be easy but would be amazing :rocket:. Different styles might have been adopted to print e.g. matrices, etc., but we may reach a consensus. And it would definitely make the printed messages consistent: the state is now quite inconsistent, with whitespaces inserted manually (for some visual alignment of the code, and whose impact when printed is maybe unsure), `=` vs. `:`, etc. :pensive:.